packages/manager/modules/pci/src/projects/project/storages/containers/containers.controller.js

```diff
@@ -1,12 +1,47 @@
+import set from 'lodash/set';
+
 export default class PciStoragesContainersController {
   /* @ngInject */
-  constructor($translate, CucCloudMessage) {
+  constructor($translate, $http, CucCloudMessage) {
     this.$translate = $translate;
+    this.$http = $http;
     this.CucCloudMessage = CucCloudMessage;
   }
 
   $onInit() {
     this.loadMessages();
+    this.publicToggleLoading = false;
+  }
+
+  onPublicToggle(container) {
+    set(container, 'publicToggleLoading', true);
+    return this.$http
+      .put(`/cloud/project/${this.projectId}/storage/${container.id}`, {
+        containerType: container.publicState ? 'private' : 'public',
```

Comment:
If there is a publicState then the container is `private` ? Not really intuitive 
Also maybe it could be handled by the model